pkg/utils/clients/factories.go

```diff
@@ -71,3 +78,16 @@ func NewHTTPClient(_ context.Context, maxConnsPerHost MaxConnsPerHost) *nethttp.
 		},
 	}
 }
+
+func authChecker(authType authcheck.AuthType) func(w nethttp.ResponseWriter, req *nethttp.Request) {
+	authCheck := authcheck.NewDefault(authType)
+	return func(w nethttp.ResponseWriter, req *nethttp.Request) {
+		if req.URL.Path == "/healthz" {
+			// Perform Authentication check.
+			if err := authCheck.Check(req.Context()); err != nil {
+				w.WriteHeader(nethttp.StatusInternalServerError)
+				return
+			}
```

Comment:
Do we need to write OK code?

Reply:
Yes, we need. Forget to add that.... Thanks